lib/Echidna/Transaction.hs

```diff
@@ -224,3 +229,36 @@ setupTx (Tx c s r g gp v (t, b)) = liftSH . sequence_ $
       SolCalldata cd  -> loadContract r >> state . calldata .= cd
     encode (n, vs) = abiCalldata
       (encodeSig (n, abiValueType <$> vs)) $ V.fromList vs
+
+saveTxs :: Maybe FilePath -> [[Tx]] -> IO ()
+saveTxs (Just d) txs = mapM_ (\v -> do let fn = d ++ "/" ++ (show . hash . show) v ++ ".txt"
+                                       b <- doesFileExist fn
+                                       unless b $ encodeFile fn (sv v)
+                             ) txs
+                             where sv = toJSON
+saveTxs Nothing  _   = return ()
+
+listDirectory :: FilePath -> IO [FilePath]
+listDirectory path = Prelude.filter f <$> getDirectoryContents path
+  where f filename = filename /= "." && filename /= ".."
+
+withCurrentDirectory :: FilePath  -- ^ Directory to execute in
+                     -> IO a      -- ^ Action to be executed
```

Comment:
This function matches the one in the [directory package](https://hackage.haskell.org/package/directory-1.3.6.0/docs/System-Directory.html#v:withCurrentDirectory), which we cannot use because it requires to upgrade the resolver: when we upgrade the resolver, we can just remove the implementation and it will work. Are you sure you want to change it?

Reply:
in that case let's leave it as is then